homeassistant/components/nuimo.py

```diff
@@ -0,0 +1,108 @@
+"""
+components.nuimo
+~~~~~~~~~~~~~~~~~~
+Component that connects to a Bluetooth reachable Nuimo device.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/nuimo/
+"""
+from logging import getLogger
+import threading
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.helpers import validate_config
+
+REQUIREMENTS = [
+    'https://github.com/shaftoe/nuimo-raspberrypi-demo/archive/'
+    'pip.zip#nuimo_core==0.0.1',
+    'https://github.com/IanHarvey/bluepy/archive/master.zip#bluepy==1.0.3']
+
+_LOGGER = getLogger(__name__)
+
+DOMAIN = "nuimo"
+EVENT_NUIMO = "nuimo_action"
+
+
+class NuimoDelegate(object):  # pylint: disable=too-few-public-methods
+    """ Translate Nuimo bluetooth messages into HASS events """
+
+    def __init__(self, hass, nuimo):
+        self._nuimo = nuimo
+        self._hass = hass
+
+    # pylint: disable=invalid-name
+    def handleNotification(self, c_handle, data):
+        """ listen for notifications and send them into hass.bus """
+        if int(c_handle) == self._nuimo.characteristicValueHandles['BATTERY']:
```

Comment:
A sensor entity is an entity that provides read only data to HA. It is visible in the frontend as a badge. In this case it could be the battery % of your nuimo.